src/modules/navigator/mission.cpp

```diff
@@ -1497,7 +1497,6 @@ Mission::generate_waypoint_from_heading(struct position_setpoint_s *setpoint, fl
 		1000000.0f,
 		&(setpoint->lat),
 		&(setpoint->lon));
-	setpoint->alt = _navigator->get_global_position()->alt;
```

Comment:
It only does this when commanded in a mission, so yes.
But it doesnt really matter, it's just to avoid a quick spike in the pitch setpoint.